cmd/erasure-server-pool.go

```diff
@@ -47,6 +47,72 @@ import (
 	"github.com/minio/pkg/v3/wildcard"
 )
 
+// MultipartUpload represents multipart upload details to be cached
+type MultipartUpload struct {
+	Bucket   string
+	Object   string
+	UploadID string
+}
+
+// MultipartUploadCacheEntry represents one multipart upload cache entry
+type MultipartUploadCacheEntry struct {
+	Upload    MultipartUpload
+	Timestamp time.Time
+}
+
+// MultipartUploadCache represents a in memory cache to hold multipart upload details
+type MultipartUploadCache struct {
+	// Key would be uploadID as it's unique UUID
+	data  map[string]MultipartUploadCacheEntry
+	mutex sync.RWMutex
+}
+
+// NewMultipartUploadsCache initializes and returns a new Cache
+func NewMultipartUploadsCache() *MultipartUploadCache {
+	cache := &MultipartUploadCache{
+		data: make(map[string]MultipartUploadCacheEntry),
+	}
+	// Start the cleanup process
+	go cache.cleanExpiredEntries()
+	return cache
+}
+
+// Set adds a value to the cache with the current timestamp
+func (c *MultipartUploadCache) Set(key string, value MultipartUpload) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	c.data[key] = MultipartUploadCacheEntry{
+		Upload:    value,
+		Timestamp: time.Now(),
+	}
+}
+
+// Get retrieves a value from the cache if it has not expired
+func (c *MultipartUploadCache) Get(key string) (MultipartUpload, bool) {
```

Comment:
```suggestion
func (c *MultipartUploadCache) Get(key string) (MultipartUpload, bool) {
```

Add another helper called `Has(key string) (bool) {`